plugins/services/src/js/containers/services/ServicesTable.js

```diff
@@ -391,6 +392,22 @@ class ServicesTable extends React.Component {
     );
   }
 
+  renderInstances(prop, service) {
+    const instancesCount = service.getInstancesCount();
```

Comment:
You could return early if instancesCount is not a number as there's no need to perform any of the other actions in that case. Similar to the following sketch: 

```js
renderInstances(prop, service) {
    const instancesCount = service.getInstancesCount();
    
    if(!Number.isInteger(instancesCount)){
       return <span>\u2014</span>;
    } 
​
    const tasksRunning = service.getTaskCount();
​
    return (
      <span>
        { 
           tasksRunning === instancesCount 
              ? ` ${tasksRunning}` 
               : ` ${tasksRunning}/${instancesCount}`
         }
      </span>
    );
  }
```

Reply:
I appreciate the feedback, I do rather see one return per method/functions because give us the impression of being cleaner with one flow because there is no use if condition (`if`)

Reply:
Why do you prefer to see only one return per method/function? Early returns help maintain readability esp. if you have a large number _preconditions_ as you don't need to nest conditions. Compare the following to examples:

```
if A is defined
  if B  is defined
    if  C  is defined
        // do something
    else 
      handle undefined C
  else 
     handle undefined B
else 
  handle undefined A
```

```
if A is undefined
  handle undefined A; 
  return;
if B is undefined
  handle undefined B; 
  return;
if C is  undefined
  handle undefined C; 
  return;

// do something
```

Please note that using early returns instead of nested conditions does't effect the cyclomatic complexity but improves readability. BTW: A ternary also affects the complexity and based on the metric (e.g., McGabe or NPATH) you're using it's even worse than a condition. 

Anyway, I'm ok with the current implementation and only wanted to provide an example of a different solution for educational purposes.